sdgym/datasets.py

```diff
@@ -45,9 +47,24 @@ def _get_dataset_path(dataset, datasets_path, bucket=None, aws_key=None, aws_sec
     return dataset_path
 
 
-def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None):
+def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None,
+                 max_columns=None):
     dataset_path = _get_dataset_path(dataset, datasets_path, bucket, aws_key, aws_secret)
-    metadata = Metadata(str(dataset_path / 'metadata.json'))
+    metadata_file = open(str(dataset_path / 'metadata.json'))
+    metadata_content = json.load(metadata_file)
+    if max_columns:
```

Comment:
Can we move all of this inside a dedicated function?

Also, since relationships can be badly broken if we cut the number of columns, I would say that we should support this only for `single-table` and `time-series` datasets. Can we raise a `ValueError` if `max_columns` is defined and the dataset has more than one table?